include/godot_cpp/classes/bindings.hpp

```diff
@@ -0,0 +1,51 @@
+/**************************************************************************/
+/*  bindings.hpp                                                          */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef GODOT_MEMBER_BINDINGS_HPP
+#define GODOT_MEMBER_BINDINGS_HPP
+
+#define GDBIND(m_class, m_name, m_properties) 	\
+const char* getter_##m_name = "get_" #m_name;			\
+const char* setter_##m_name = "set_" #m_name;			\
```

Comment:
These shouldn't be variables IMO it pollutes and could just be as is, it doesn't make the code any simpler as you still need to write `getter_##m_name` that's the same as just `"get_" #m_name`

Reply:
Yeah to be honest I thought the same as well but when I tried just adding it as `D_METHOD("get_" #m_name)` the compiler was complaining that it couldn't find the method. However, I've since changed it to something along the lines of: 

```
#define GDBIND(m_class, m_name, m_properties, m_get_method, m_set_method) /*******************************************************/ \
::godot::ClassDB::bind_method(D_METHOD(#m_get_method), &m_class::m_get_method);	                                                    \
::godot::ClassDB::bind_method(D_METHOD(#m_set_method, "p_temp"), &m_class::m_set_method);	                                        \
::godot::ClassDB::add_property(get_class_static(), m_properties, m_set_method, m_get_method);                                       \
```
Was this more of the format you were thinking of?

Reply:
I'm thinking that the method in `D_METHOD` and that in the class itself should be allowed to differ, that's common with some cases and naming standards (also parameters to the bound function shouldn't have a `p_` prefix by the naming standards)

But it quickly becomes very complicated and fragile IMO, I personally don't like hiding relevant information that matters to the user behind a macro, it hurts readability